osquery/tables/system/linux/cpu_time.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+#include <vector>
+#include <boost/filesystem.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+#include <osquery/filesystem.h>
+
+#include <osquery/core/conversions.h>
+
+namespace osquery {
+namespace tables {
+
+const std::string kProcStat = "/proc/stat";
+
+std::vector<std::string> procFromFile(const std::string path) {
+    std::string content;
+    std::vector<std::string> proc_lines;
+
+    if (!isReadable(path).ok()) {
+        return proc_lines;
+    }
+
+    if (!forensicReadFile(path, content).ok()) {
+        return proc_lines;
+    }
+
+    auto lines = split(content, "\n");
+
+    for (auto& line : lines) {
+        boost::trim(line);
+        if (boost::starts_with(line, "cpu")) {
+            proc_lines.push_back(line);
+        }
+    }
+
+    // Remove first cpu line which doesn't give specific core information
+    proc_lines.erase(proc_lines.begin());
+
+    return proc_lines;
+}
+
+Row genCpuTimeLine(std::string line) {
+    Row r;
+
+    std::vector<std::string> words;
+    split(words, line, boost::is_any_of(" "));
+
+    r["core"] = words[0].back();
+    r["user"] = words[1];
+    r["nice"] = words[2];
+    r["system"] = words[3];
+    r["idle"] = words[4];
+    r["iowait"] = words[5];
+    r["irq"] = words[6];
+    r["softirq"] = words[7];
+    r["steal"] = words[8];
+    r["guest"] = words[9];
+    r["guest_nice"] = words[10];
+
+    return r;
+}
+
+QueryData genCpuTime(QueryContext &context) {
+    QueryData results;
+
+    auto proc_lines = procFromFile(kProcStat);
+    for (auto line : proc_lines) {
+        results.push_back(genCpuTimeLine(line));
```

Comment:
With the internal `gen....` functions we usually pass the results structure as a reference. Then if there are problems generating a row you don't have to return a value and thus push an invalid row into the results.
Try:
```
genCpuTimeLine(line, results);
```

Then modify the function signature to:
```
static void genCpuTimeLine(const std::string& line, QueryData& results);
```

When you have a valid row (aka, the size of the split line == 11) you can `results.push_back(r);`